spring-cloud-core/src/main/java/org/springframework/cloud/service/common/DB2ServiceInfo.java

```diff
@@ -0,0 +1,27 @@
+package org.springframework.cloud.service.common;
+
+import org.springframework.cloud.service.ServiceInfo;
+
+@ServiceInfo.ServiceLabel("db2")
+public class DB2ServiceInfo extends RelationalServiceInfo {
+
+    private static final String JDBC_URL_TYPE = "db2";
+
+    public static final String DB2_SCHEME = JDBC_URL_TYPE;
+
+	public DB2ServiceInfo(String id, String url) {
+		super(id, url, JDBC_URL_TYPE);
+	}
+
+	@Override
+	public String getJdbcUrl() {
+		if (getUriInfo().getUriString().startsWith(JDBC_PREFIX)) {
+			return getUriInfo().getUriString();
+		}
+		                    
+		return String.format("jdbc:%s://%s:%d/%s",
```

Comment:
@scottfrederick DB2 JDBC driver does not support passing the username and password in the URL (either like jdbc:db2://username:password@... or jdbc:db2://.../dbname?user=...&password=...). It is required to pass it as a separate properties or separate string arguments to the DriverManager.connect(). This is why i didn't include them in the JDBC URL